### PR TITLE
Add documentation for React Native tests

### PR DESCRIPTION
**Pull request by SnehPandya18**

### CODE REVIEW

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]
The Codemagic instance types use underscores not hyphens so "mac-mini or mac-pro for Macintosh instances" should be "mac_mini or mac_pro for macOS instances". Also use "macOS" instead of "Macintosh"

File: content/testing-yaml/testing.md [hunk 1]
```diff
@@ -11,12 +11,171 @@ Test scripts are added under `scripts` in the [overall architecture](../getting-
...
+In the root directory of the project, add a new file and name it `jest.config.js`. Within `jest.config.js`, add the below code:
+
+```javascript
+module.exports = {
+  preset: 'react-native',
+  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
+};
+```
+
+`present` is a preset that is used as a base for Jest’s configuration. A preset should point to an npm module that has a `jest-preset.json` or `jest-preset.js` file at the root.
```
Comment (helinanever) [inline content/testing-yaml/testing.md:38]: present > preset

File: content/testing-yaml/testing.md [hunk 2]
```diff
@@ -11,12 +11,171 @@ Test scripts are added under `scripts` in the [overall architecture](../getting-
...
+```
+
+Once the project setup is done on Codemagic, you need to add the testing configuration in the `codemagic.yaml` file as shown below for the tests to be executed:
+
+```yaml
+workflows:
+  
+  my-workflow:
+    name: <your workflow name>
+      instance_type: linux                # specify linux or linux_2 for Linux instances; mac_mini or mac_pro for MacOS instances
```
Comment (helinanever) [inline content/testing-yaml/testing.md:72]: MacOS > macOS

**Comment by SnehPandya18** [comment]
> The Codemagic instance types use underscores not hyphens so "mac-mini or mac-pro for Macintosh instances" should be "mac_mini or mac_pro for macOS instances". Also use "macOS" instead of "Macintosh"

The request changes are updated in the latest commit. Please check.

File: content/testing-yaml/testing.md [hunk 3]
```diff
@@ -11,12 +11,171 @@ Test scripts are added under `scripts` in the [overall architecture](../getting-
...
+## React Native Integration Tests using Appium and Emulator
+
+[Appium](https://appium.io/) is an open-source testing automation framework for testing cross-platform applications and mobile applications. You can use Appium and WebDriverIO with React Native thanks to its out-of-the-box support.
+
+You need to install and setup WebDriverIO in your project root directory before running CI or tests on Codemagic. On your local machine, run the below command and provide input to a series of questions:
+
+```bash
+npx wdio config
+```
+Once it done, `wdio.conf` file will be generated inside the `tests` directory.
```
Comment (helinanever) [inline content/testing-yaml/testing.md:105]: Once it done > Once it's done

File: content/testing-yaml/testing.md [hunk 4]
```diff
@@ -11,12 +11,171 @@ Test scripts are added under `scripts` in the [overall architecture](../getting-
...
+```
+
+Once the project setup is done on Codemagic, you need to add the testing configuration in the `codemagic.yaml` file as shown below for the tests to be executed:
+
+```yaml
+workflows:
+  
+  my-workflow:
+    name: <your workflow name>
+      instance_type: linux                # specify linux or linux_2 for Linux instances; mac_mini or mac_pro for MacOS instances
```
Comment (helinanever) [inline content/testing-yaml/testing.md:149]: MacOS > macOS

**Review by helinanever — CHANGES_REQUESTED** [review]
Minor spelling issues

**Comment by SnehPandya18** [comment]
> Minor spelling issues

Thanks for pointing out. The spellings are corrected.

**Review by helinanever — APPROVED** [review]
LGTM

**Review by nevercode-kevin — APPROVED** [review]

**Pull request merged**